### PR TITLE
1059 plugin development

### LINKED ISSUE #1059: design and document plugin development workflow
Extract GDC plugin from the monorepo, publish plugin-development-tools package on npm. Get GDC plugin running outside of the monorepo.

Then lay out and document the process for somebody to develop a plugin in an external repository.

Something like: 

* make a repository from the boilerplate repository
* run a dev server for your plugin
* make a test configuration for developing the plugin, that lives inside the plugin repo
* run a local jbrowse-web dev server, and point the jbrowse-web at the plugin's development configuration

### PR DESCRIPTION
**Pull request by rbuels**
* moves the GDC plugin out into its own repo
* started npm publishing @gmod/jbrowse-core and @gmod/jbrowse-developer-tools
* modules in the ReExports list are now provided to plugins as window.JBrowseExports['module_name'], which is consumable by plugins using [webpack externals](https://webpack.js.org/configuration/externals/), which is part of the base plugin configuration  

fixes #1059

### CODE REVIEW

**Comment by cmdcolin** [comment]
@rbuels looks like tests pass now :)

**Comment by cmdcolin** [comment]
This has my seal of approval...any objections?

File: packages/core/package.json [hunk 1]
```diff
@@ -13,16 +13,14 @@
...
-  ],
   "scripts": {
     "test": "jest",
     "coverage": "npm test -- --coverage",
     "lint": "tsc --noEmit && eslint --ext .js,.ts,.jsx,.tsx .",
     "clean": "rimraf dist",
     "prebuild": "npm run clean",
-    "build": "babel . --root-mode upward --out-dir dist --extensions '.ts,.js'",
+    "prepare": "npm run build",
+    "build": "NODE_ENV=production babel . --root-mode upward --out-dir dist --extensions '.ts,.js,.tsx,.jsx' && cp package.json CHANGELOG.md LICENSE dist/",
```
Comment (cmdcolin) [inline packages/core/package.json:23]: what is the purpose of this "copying files in to the dist folder"?

**Comment by rbuels** [comment]
That is so that the distribution is tree shakable, so people can import
single is files rather than just go from the index. Material ui icons does
a similar thing but much more complicated

**Pull request merged**